src/odm2_postgres_api/queries/user.py

```
@@ -0,0 +1,80 @@
+import json
+from base64 import b64decode
+from datetime import datetime
+from typing import List, Tuple, Optional
+
+import asyncpg
+from pydantic import BaseModel
+
+from odm2_postgres_api.schemas.schemas import People, Affiliations, PersonExternalIdentifiers
+
+
+class NivaPortUser(BaseModel):
+    """
+    User object passed from niva-port as a base64 encoded json string
+    """
+    id: str
+    email: str
+    roles: List[str]
+    displayName: str
+
+
+def full_name_to_split_tuple(fullname: str) -> Tuple[str, Optional[str], str]:
+    head, *mid, tail = fullname.strip().split(" ")
```

if there is only 1 name in fullname this will throw a valueerror and the error handling on line 26 will never be hit:
alternative1:
```python
    try:
        head, *mid, tail = fullname.strip().split(" ")
        middle_name = " ".join(mid) if mid else None
        name_tuple = (head, middle_name, tail)
        return name_tuple
    except ValueError:  # destructuring can lead to ValuErrors if there are not enough names in the string
        raise Exception("could not parse user fullname")
```
alt2
```python
    names = fullname.strip().split(" ")
    if len(names) == 1:
        raise Exception(f"Need at least first and last name: {names=}")
    if len(names) == 2:
        return names[0], None, names[-1]
    else:
        return names[0], " ".join(names[1:-1]), names[-1]
```

---

also, hitting that error conditions has to be part of one of the tests

---

Good catch.

I think since people.personfirstname and people.personlastname are both required fields in odm2, I'd rather throw an error if we'd get just one name. Most likely this is an error somewhere else in the system, and I'd prefer it to blow up.

I added a check for this and throw ValueError (what is typical exception to throw when a function receives bad input?)

PS: I am not printing the name as part of exception, I think we should avoid that due to GDPR restrictions. I think the "right to be deleted" can be a hassle if we end up having names lying around in our logs. That would probably be taken care of by log retention policies, but still..